src/main/java/com/lgtm/easymoney/services/impl/TransferServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+package com.lgtm.easymoney.services.impl;
+
+import com.lgtm.easymoney.enums.Category;
+import com.lgtm.easymoney.enums.TransactionStatus;
+import com.lgtm.easymoney.models.Transaction;
+import com.lgtm.easymoney.models.User;
+import com.lgtm.easymoney.payload.TransactionRsp;
+import com.lgtm.easymoney.payload.TransferReq;
+import com.lgtm.easymoney.payload.TransferRsp;
+import com.lgtm.easymoney.services.TransactionService;
+import com.lgtm.easymoney.services.TransferService;
+import com.lgtm.easymoney.services.UserService;
+import org.apache.coyote.Response;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Service
+public class TransferServiceImpl implements TransferService {
+    private final TransactionService transactionService;
+    private final UserService userService;
+
+    @Autowired
+    public TransferServiceImpl(TransactionService transactionService, UserService userService) {
+        this.transactionService = transactionService;
+        this.userService = userService;
+    }
+
+    private Transaction createATransaction(User fromUser, User toUser, BigDecimal amount, Category category, String desc) {
+        Transaction transaction = new Transaction();
+        transaction.setFrom(fromUser);
+        transaction.setTo(toUser);
+        transaction.setAmount(amount);
+        transaction.setCategory(category);
+        transaction.setDescription(desc);
+        transaction.setStatus(TransactionStatus.REQ_PENDING);
```

Comment:
TL;DR: status is not proper here

so the thing is: requests are `pending -> complete/denied`, and i think we should also follow the same route for transfers, even if **transfers should be successful without receiver's actions**.

in that way, if transfers failed to execute we can have a status check in between.

cc: @howieraem

Reply:
In transfer, this REQ_PENDING status is not exposed to user. This is only something that I have to set before calling executeTransaction based on the current implementation of transacationService. Not a big deal but can be improved if there's a better way to handle the status stuff for both request and transfer.

Reply:
Probably we can make all the status TRANS_*. In this way,, REQ_PENDING becomes TRANS_PENDING, and this makes sense for both request and transfer. After all, transfer is the same as request except that the change from PENDING to COMPLETE/DENIED is determined directly by our backend logic rather than determined by user. cc: @howieraem

Reply:
this sounds good, how to search all requests then?

Reply:
Finalized modification:
1. REQ_* renamed to TRANS_*
2. update findByFromOrToAndStatus(User from, User To, Status s)